### PR TITLE
Rope test

### PR DESCRIPTION
**Pull request by philipandag**
Basing on: https://github.com/Liberum-Neto/liberum-neto/pull/53, merge after it

### CODE REVIEW

File: tests/publish-download-rope-net.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+# wait for nodes to connect
+sleep 0.5
+
+# create and provide file
+echo "${FILE_CONTENT}" > "$FILE_NAME"
+$CLI_BIN -d publish-file ${N_NAMES[0]} "$FILE_NAME" 2> /dev/null
+
+init_asserts
+
+RESULT=$($CLI_BIN -d download-file ${N_NAMES[$((variable - 1))]} "${BLAKE3_HASH}" 2> /dev/null)
```
Comment (originalmk) [inline tests/publish-download-rope-net.sh:62]: Why `variable`?

**Review by originalmk — CHANGES_REQUESTED** [review]

File: tests/publish-download-rope-net.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+NODE_ADDR_SUFFIX="/quic-v1"
+
+echo "Provide and download file test:"
+
+# run daemon
+killall liberum_core &> /dev/null
+$CORE_BIN --daemon  &> /dev/null &
+sleep 0.1; # the socket file is created asynchronously and may not be ready yet :))))
+
+# create ndoes
```
[the remark this answers is not in the document]
Comment (philipandag) [reply tests/publish-download-rope-net.sh]: https://github.com/Liberum-Neto/liberum-neto/pull/54/commits/2087f73a75badde39cd115b0bc167a3cda83c7e3

File: tests/publish-download-rope-net.sh (hunk 1 above)
Comment (philipandag) [reply tests/publish-download-rope-net.sh]: https://github.com/Liberum-Neto/liberum-neto/pull/54/commits/5ed944007de4269ee0e844114b957e35ca73155e

File: tests/lib/asserts.sh
[the remark this answers is not in the document]
Comment (philipandag) [reply tests/lib/asserts.sh:1]: It's always set by run-tests.sh

**Review by originalmk — APPROVED** [review]

**Pull request merged**